### PR TITLE
anki: Enable tests

### PR DESCRIPTION
**Pull request by adisbladis**
###### Motivation for this change
Enable tests when building `anki`

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by matthewbauer — APPROVED** [review]

**Comment by orivej** [comment]
Would you like to convert this into `buildPythonApplication` using `pythonPackages.callPackage`, replace inherit with function arguments, remove the override of the list of phases, replace pythonPath with propagatedBuildInputs, and use checkInputs for the new dependencies?

File: pkgs/games/anki/default.nix [hunk 1]
```diff
@@ -31,9 +32,9 @@ stdenv.mkDerivation rec {
     pythonPath = [ pyqt4 sqlalchemy pyaudio beautifulsoup httplib2 ]
               ++ lib.optional plotsSupport matplotlib;
 
-    buildInputs = [ python wrapPython lame mplayer libpulseaudio ];
+    buildInputs = [ python wrapPython lame mplayer libpulseaudio pytest glibcLocales nose ];
 
-    phases = [ "unpackPhase" "patchPhase" "installPhase" ];
+    phases = [ "unpackPhase" "patchPhase" "testPhase" "installPhase" ];
```
Comment (Mic92) [inline pkgs/games/anki/default.nix]: if the `fixupPhase` does not do any harm here, please remove the explicit phases selection here.

**Comment by adisbladis** [comment]
@orivej I thought about that too but didn't really want to since I don't use the package myself. Anyway, it has been done.

**Comment by orivej** [comment]
Thank you! BTW the difference between `buildPythonPackage` and `buildPythonApplication` is that the latter does not prepend python version to the name of the package. I've fixed it in a2bebdd6cedb58819cc1738282b9f02964d34a9a.

**Pull request merged**